Deploy-CAM.ps1

```diff
@@ -2924,58 +2924,101 @@ else {
     do {
         if ( -not $domainName ) {
             if( -not $deployOverDC ) {
-                $domainNameMessage = "Please enter new fully qualified domain name of the domain which will be created, including a '.' such as example.com"
+                $domainNameMessage = "`nPlease enter new fully qualified domain name of the domain which will be created, including a '.' such as example.com"
             }
             else {
-                $domainNameMessage = "Please enter the fully qualified domain name of the domain to connect to. The name must include a '.' such as example.com"
+                $domainNameMessage = "`nPlease enter the fully qualified domain name of the domain to connect to. The name must include a '.' such as example.com"
             }
             Write-Host $domainNameMessage
             $domainName = Read-Host "Domain name"
         }
 
-        # https://social.technet.microsoft.com/Forums/scriptcenter/en-US/db2d8388-f2c2-4f67-9f84-c17b060504e1/regex-for-computer-fqdn?forum=winserverpowershell
-        if (-not $($domainName -imatch '(?=^.{1,254}$)(^(?:(?!\d+\.|-)[a-zA-Z0-9_\-]{1,63}(?<!-)\.?)+(?:[a-zA-Z]{2,})$)')) {
-            Write-Host "The domain name must include two or more components separated by a '.'"
+        if (-not $($domainName -match '^[a-zA-Z0-9]{1,61}[a-zA-Z0-9](?:\.[a-zA-Z]{2,})+$')) {
```

Comment:
This definitely looks more straightforward than the previous regex but I have some concerns here:

- even though the `match` operator is case-insensitive, we should be explicit and use the `imatch` operator; this will also remove the need to account for uppercase characters in the regex
- `0-9` can be replaced with the digit class `\d`
- a hyphen (`-`) is a valid character in a domain name, but this regex doesn't allow it

Reply:
This regex is so complex it does hurt my brain... @ivpetrov87 found the original one from some online thing. Does the new one work with 3-level domains such as blah.blah.com ? I don't know what my pereference is here but the 'minimal change' part of me says that maybe what we want is the original regex plus a search for 'there must be a dot somewhere' to force a two-level domain.

Reply:
Add another checking for dot only and use original regex